mne/preprocessing/ica.py

```diff
@@ -2093,3 +2110,228 @@ def _band_pass_filter(ica, sources, target, l_freq, h_freq, verbose=None):
+        ttl = 'Template from subj. {0}'.format(str(template[0]))
+        template_fig = icas[template[0]].plot_components(picks=template[1],
+                                                         ch_type=ch_type,
+                                                         title=ttl)
+        template_fig.subplots_adjust(top=0.8)
+        template_fig.canvas.draw()
+
+    # first run: use user-selected map
+    if isinstance(threshold, (int, float)):
+        try:
+            nt, mt, s, mx = find_max_corrs(all_maps, target, threshold)
+        except ValueError:
+            logger.info("No component detected using find_outliers."
+                        " Consider using threshold='auto'")
+            return icas
+    elif len(threshold) > 1:
+        paths = [find_max_corrs(all_maps, target, t) for t in threshold]
+        # find iteration with highest avg correlation with target
+        nt, mt, s, mx = paths[np.argmax([path[2] for path in paths])]
+
+    # second run: use output from first run
+    if isinstance(threshold, (int, float)):
+        try:
+            nt, mt, s, mx = find_max_corrs(all_maps, nt, threshold)
+        except ValueError:
+            if threshold > 1:
+                logger.info("No component detected using find_outliers. "
+                            "Consider using threshold='auto'")
+            return icas
+    elif len(threshold) > 1:
+        paths = [find_max_corrs(all_maps, nt, t) for t in threshold]
+        # find iteration with highest avg correlation with target
+        nt, mt, s, mx = paths[np.argmax([path[1] for path in paths])]
+
+    nones = list(range(len(icas)))
+    allmaps, indices, subjs = [], [], []
+    logger.info("Median correlation with constructed map: %0.3f" % mt)
+    if plot:
+        logger.info("Displaying selected ICs per subject.")
+
+    for ii, (ica, max_corr) in enumerate(zip(icas, mx)):
+        if (label is not None) and (not hasattr(ica, 'labels_')):
+            ica.labels_ = {}
+        if len(max_corr) > 0:
+            if isinstance(max_corr[0], np.ndarray):
+                max_corr = max_corr[0]
+            if label is not None:
+                ica.labels_[label] = list(set(list(max_corr) +
+                                          ica.labels_.get(label, [])))
+            if plot is True:
+                allmaps.extend(get_ica_map(ica, components=max_corr))
+                subjs.extend([ii] * len(max_corr))
+                indices.extend(max_corr)
+            nones.remove(ii)
+
+    if plot is True:
+        labelled_ics = _plot_corrmap(allmaps, subjs, indices, ch_type, ica,
+                                     label)
+
+    if nones:
+        logger.info("No maps selected for subject(s) " +
+                    ", ".join([str(x) for x in nones]) +
+                    ", consider a more liberal threshold.")
+    else:
+        logger.info("At least 1 IC detected for each subject.")
+
+    return None if plot is False else (template_fig, labelled_ics)
```

Comment:
One problem with using `if plot is True` above and `if plot is False` here, is that you cannot assume they are complementary, e.g. `2 is True` is `False` and `2 is False` is `False`. It should be safer to just use `if plot:` above and `if not plot` here, because those are complementary.